src/ListView.php

```diff
@@ -141,72 +116,159 @@ public function renderItems()
-     * @param int   $index the zero-based index of the data model in the model array returned by [[dataProvider]].
-     *
+     * @param mixed $key the key value associated with the data model
+     * @param int $index the zero-based index of the data model in the model array returned by [[dataProvider]].
      * @return string the rendering result
+     * @throws \Throwable
      */
-    public function renderItem($model, $key, $index)
+    public function renderItem($model, $key, $index): string
     {
         if ($this->itemView === null) {
             $content = $key;
         } elseif (is_string($this->itemView)) {
-            $content = $this->getView()->render($this->itemView, array_merge([
-                'model'  => $model,
-                'key'    => $key,
-                'index'  => $index,
-                'widget' => $this,
-            ], $this->viewParams));
-        } else {
+            $content = $this->getView()->render(
+                $this->getAliases()->get($this->getItemViewPath()),
+                array_merge(
+                    [
+                        'model' => $model,
+                        'key' => $key,
+                        'index' => $index,
+                        'widget' => $this,
+                    ],
+                    $this->viewParams
+                ),
+                $this
+            );
+        } elseif (is_callable($this->itemView)) {
             $content = call_user_func($this->itemView, $model, $key, $index, $this);
+        } else {
+            // TODO make InvalidConfigException
+            throw new \RuntimeException('Unknown type of $itemView');
         }
         if ($this->itemOptions instanceof Closure) {
             $options = call_user_func($this->itemOptions, $model, $key, $index, $this);
         } else {
             $options = $this->itemOptions;
         }
+
         $tag = ArrayHelper::remove($options, 'tag', 'div');
-        $options['data-key'] = is_array($key) ? json_encode($key, JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE) : (string) $key;
+        $options['data-key'] = is_array($key) ? json_encode(
+            $key,
+            JSON_THROW_ON_ERROR | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE
+        ) : (string)$key;
 
         return Html::tag($tag, $content, $options);
     }
+
+    public function withItemOptions($itemOptions): self
+    {
+        if (is_array($itemOptions)) {
+            $this->itemOptions = ArrayHelper::merge($this->itemOptions, $itemOptions);
+
+            return $this;
+        }
+
+        if (is_callable($itemOptions)) {
+            $this->itemOptions = $itemOptions;
+
+            return $this;
+        }
+
+        throw new \InvalidArgumentException();
+    }
+
+    public function getItemViewPath(): ?string
+    {
+        if (is_callable($this->itemView)) {
+            return ($this->itemView)();
+        }
+
+        return $this->itemView;
+    }
+
+    /**
+     * @inheritDoc
```

Comment:
No need for inheritDoc in case it's the only phpdoc comment without any changes or additions.